toasty/image.py

```diff
@@ -854,24 +842,44 @@ def asarray(self):
             self._array = np.asarray(self._pil)
         return self._array
 
-    def aspil(self):
+    def aspil(self, pixel_cut_low=None, pixel_cut_high=None):
         """Obtain the image data as :class:`PIL.Image.Image`.
 
+        Parameters
+        ----------
+        pixel_cut_low : number or ``None`` (the default)
+            An optional value used to stretch the pixel values to the new range
+            as defined by pixel_cut_low and pixel_cut_high. Only used if the
+            image was not loaded as a PIL image, and must be used together with
+            pixel_cut_high. To get reasonable values for arrays with dtype other
+            than ``unit8`` this parameter is highly recommended.
+        pixel_cut_high : number or ``None`` (the default)
+            An optional value used to stretch the pixel values to the new range
+            as defined by pixel_cut_low and pixel_cut_high. Only used if the
+            image was not loaded as a PIL image, and must be used together with
+            pixel_cut_low. To get reasonable values for arrays with dtype other
+            than ``unit8`` this parameter is highly recommended.
         Returns
         -------
         If the image was loaded as a PIL image, the underlying object will be
         returned. Otherwise the data array will be converted into a PIL image,
         which requires that the array have an RGB(A) format with a shape of
-        ``(height, width, planes)``, where ``planes`` is 3 or 4, and a dtype of
-        ``uint8``.
+        ``(height, width, planes)``, where ``planes`` is 3 or 4.
 
         """
         if self._pil is not None:
             return self._pil
-        if self.mode.try_as_pil() is None:
-            raise Exception(
-                f"Toasty image with mode {self.mode} cannot be converted to PIL"
-            )
+        if (
+            pixel_cut_low is not None
+            and pixel_cut_high is not None
+            and pixel_cut_low != pixel_cut_high
+        ):
+            # TODO don't operate on potential alpha channel
+            array = (self._array - pixel_cut_low) / (
+                pixel_cut_high - pixel_cut_low
+            ) * 255 + 0.5
+            array = np.uint8(np.clip(array, 0, 255))
```

Comment:
As for your question, I'm not aware of a cleaner way to do this other than indexing the arrays as something like `array[...,:3]` to exclude the alpha channel in RGBA format.